### ISSUE TITLE
Uncaught Error: Cannot call function 'getPosition' on missing remote object 28Error: Cannot call...

### ISSUE THREAD

**Issue by KehindeAyanleye** [atom/atom#16630]
**Atom**: 1.23.3 x64
**Electron**: 1.6.15
**OS**: Mac OS X 10.13.2
**Thrown From**: Atom Core

### Stack Trace

Uncaught Error: Cannot call function 'getPosition' on missing remote object 28
Error: Cannot call function 'getPosition' on missing remote object 28
    at throwRPCError (/Applications/Atom.app/Contents/Resources/electron.asar/browser/rpc-server.js:143:17)
    at EventEmitter.<anonymous> (/Applications/Atom.app/Contents/Resources/electron.asar/browser/rpc-server.js:354:7)
    at emitMany (events.js:127:13)
    at EventEmitter.emit (events.js:201:7)
    at WebContents.<anonymous> (/Applications/Atom.app/Contents/Resources/electron.asar/browser/api/web-contents.js:256:37)
    at emitTwo (events.js:106:13)
    at WebContents.emit (events.js:191:7)

```
At /Applications/Atom.app/Contents/Resources/electron.asar/renderer/api/remote.js:234

Error: Cannot call function 'getPosition' on missing remote object 28
Error: Cannot call function 'getPosition' on missing remote object 28
    at throwRPCError (/Applications/Atom.app/Contents/Resources/electron.asar/browser/rpc-server.js:143:17)
    at EventEmitter.<anonymous> (/Applications/Atom.app/Contents/Resources/electron.asar/browser/rpc-server.js:354:7)
    at emitMany (events.js:127:13)
    at EventEmitter.emit (events.js:201:7)
    at WebContents.<anonymous> (/Applications/Atom.app/Contents/Resources/electron.asar/browser/api/web-contents.js:256:37)
    at emitTwo (events.js:106:13)
    at WebContents.emit (events.js:191:7)
    at metaToValue (/Applications/Atom.app/Contents/Resources/electron.asar/renderer/api/remote.js:234:13)
    at Object.remoteMemberFunction (/Applications/Atom.app/Contents/Resources/electron.asar/renderer/api/remote.js:118:18)
    at AtomEnvironment.module.exports.AtomEnvironment.getWindowDimensions (/Applications/Atom.app/Contents/Resources/app/src/atom-environment.js:691:34)
    at AtomEnvironment.module.exports.AtomEnvironment.storeWindowDimensions (/Applications/Atom.app/Contents/Resources/app/src/atom-environment.js:725:42)
    at WindowEventHandler.handleWindowBlur (/Applications/Atom.app/Contents/Resources/app/src/window-event-handler.js:189:32)
```

### Commands

```
  3x -2:23.4.0 core:move-left (input.hidden-input)
     -2:22.6.0 core:backspace (input.hidden-input)
     -2:21.9.0 core:confirm (input.hidden-input)
     -2:17.3.0 core:move-left (input.hidden-input)
     -2:16.9.0 core:backspace (input.hidden-input)
     -2:14.5.0 core:select-all (input.hidden-input)
     -2:14.3.0 core:move-left (input.hidden-input)
     -2:13.4.0 core:confirm (input.hidden-input)
  2x -1:47.8.0 core:copy (input.hidden-input)
     -1:35.9.0 project-find:show (input.hidden-input)
     -1:34.4.0 core:paste (input.hidden-input)
  3x -1:32.7.0 core:backspace (input.hidden-input)
     -1:31.4.0 core:confirm (input.hidden-input)
     -1:23.7.0 project-find:show (input.hidden-input)
     -1:23.2.0 core:move-right (input.hidden-input)
     -1:22.8.0 core:confirm (input.hidden-input)
```

### Non-Core Packages

```
atom-eco 0.2.0 
color-picker 2.2.5 
language-slim 0.6.0 
merge-conflicts 1.4.5 
```

**Comment by rsese** [comment]
Thanks for the report!  Can you confirm a few things for us?

* what were you doing at the time of the error?
* is the error is reproducible?
* if reproducible, do you see the error in safe mode (atom --safe)?

**Comment by rsese** [comment]
From https://github.com/atom/atom/issues/16636#issue-292017574:

> Unfortunately, I probably don't have good steps to reproduce, here. I'd just edited a method in my Rails app's application helper, saved it, alt-tabbed over to a browser to test, and was greeted with this error when I alt-tabbed back.

**Comment by rsese** [comment]
From https://github.com/atom/atom/issues/16705 (Windows 10):

> select content of a file using CTRL + A
> go to another program
> go back to atom and do CTRL + C

**Comment by rsese** [comment]
From https://github.com/atom/atom/issues/16840:

> Build .scss file
> It happened only once, can't reproduce it 100% times

**Comment by borishere** [comment]
> Thanks for the report - this looks like a duplicate of #16630 so you can subscribe there for updates if you'd like. I'll go ahead and copy over your comments over but do you happen to remember if you tabbed out of Atom and then switched back when you saw the error?

Yes. It was something like:

1. Press 'build' command
2. Go to another program
3. Return back to Atom
4. An error appeared

**Comment by scsmash3r** [comment]
Can't really figure out how to reproduce that - was doing what I was doing all the time. Never had this issue for 3 years of using Atom, and it just randomly appeared today.

**Atom**: 1.40.0 x64
**Electron**: 3.1.10
**OS**: Windows 8.1
**Thrown From**: Atom Core


### Stack Trace

Uncaught Error: Cannot call function 'getPosition' on missing remote object 537
Error: Cannot call function 'getPosition' on missing remote object 537
    at throwRPCError (C:\Users\Home\AppData\Local\atom\app-1.40.0\resources\electron.asar\browser\rpc-server.js:145:17)
    at EventEmitter.<anonymous> (C:\Users\Home\AppData\Local\atom\app-1.40.0\resources\electron.asar\browser\rpc-server.js:367:7)
    at EventEmitter.emit (events.js:182:13)
    at WebContents.<anonymous> (C:\Users\Home\AppData\Local\atom\app-1.40.0\resources\electron.asar\browser\api\web-contents.js:294:13)
    at WebContents.emit (events.js:182:13)


```
At C:\Users\Home\AppData\Local\atom\app-1.40.0\resources\electron.asar\renderer\api\remote.js:219

Error: Cannot call function 'getPosition' on missing remote object 537
Error: Cannot call function 'getPosition' on missing remote object 537
    at throwRPCError (~/AppData/Local/atom/app-1.40.0/resources/electron.asar/browser/rpc-server.js:145:17)
    at EventEmitter.<anonymous> (~/AppData/Local/atom/app-1.40.0/resources/electron.asar/browser/rpc-server.js:367:7)
    at EventEmitter.emit (events.js:182:13)
    at WebContents.<anonymous> (~/AppData/Local/atom/app-1.40.0/resources/electron.asar/browser/api/web-contents.js:294:13)
    at WebContents.emit (events.js:182:13)

    at metaToException (~/AppData/Local/atom/app-1.40.0/resources/electron.asar/renderer/api/remote.js:271:17)
    at Object.exception (~/AppData/Local/atom/app-1.40.0/resources/electron.asar/renderer/api/remote.js:219:30)
    at metaToValue (~/AppData/Local/atom/app-1.40.0/resources/electron.asar/renderer/api/remote.js:223:28)
    at Object.remoteMemberFunction (~/AppData/Local/atom/app-1.40.0/resources/electron.asar/renderer/api/remote.js:123:16)
    at AtomEnvironment.getWindowDimensions (~/AppData/Local/atom/app-1.40.0/resources/app/static/<embedded>:1:744557)
    at AtomEnvironment.storeWindowDimensions (~/AppData/Local/atom/app-1.40.0/resources/app/static/<embedded>:1:744972)
    at WindowEventHandler.handleWindowBlur (~/AppData/Local/atom/app-1.40.0/resources/app/static/<embedded>:11:285354)
```

### Commands

```
     -3:06.8.0 core:copy (input.hidden-input)
     -3:03.7.0 intentions:highlight (input.hidden-input)
     -3:03.6.0 core:paste (input.hidden-input)
     -3:01.7.0 intentions:highlight (input.hidden-input)
     -3:01.6.0 core:copy (input.hidden-input)
     -3:00.9.0 intentions:highlight (input.hidden-input)
     -3:00.8.0 core:paste (input.hidden-input)
  4x -2:50 core:delete (input.hidden-input)
     -2:39.1.0 intentions:highlight (input.hidden-input)
     -2:39 core:save (input.hidden-input)
     -1:24.4.0 intentions:highlight (input.hidden-input)
     -1:24.3.0 core:copy (input.hidden-input)
     -1:23 intentions:highlight (input.hidden-input)
     -1:23 core:paste (input.hidden-input)
     -1:17.9.0 intentions:highlight (input.hidden-input)
     -1:17.7.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
ariake-dark-syntax 0.1.3 
atom-beautify 0.33.4 
atom-ide-ui 0.13.0 
atom-material-syntax 1.0.8 
atom-terminal 0.8.0 
auto-reveal-in-sidebar 0.8.0 
busy-signal 2.0.1 
docblockr 0.13.7 
file-icons 2.1.33 
fonts 3.4.0 
hey-pane 1.1.2 
hide-tree-view-icons 2.0.0 
highlight-selected 0.16.0 
ide-php 0.7.18 
intentions 1.1.5 
linter 2.3.0 
linter-eslint 8.5.5 
linter-jshint 3.1.16 
linter-php 1.6.1 
linter-ui-default 1.7.1 
minimap 4.29.9 
php-cs-fixer 4.1.1 
php-twig 4.0.0 
remember-folds 0.3.0 
symbols-list 2.5.4 
sync-settings 0.8.6 
tab-switcher 1.5.6 
todo-show 2.3.2 
toggle-quotes 1.1.4 
tree-view-autoresize 2.3.0 
vertical-tabs-plus 2.4.0 
zentabs 0.8.9 
```

**Comment by kavon** [comment]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.40.0 x64
**Electron**: 3.1.10
**OS**: Ubuntu 18.04.3
**Thrown From**: Atom Core


### Stack Trace

Uncaught Error: Cannot call function 'getPosition' on missing remote object 16
Error: Cannot call function 'getPosition' on missing remote object 16
    at throwRPCError (/usr/share/atom/resources/electron.asar/browser/rpc-server.js:145:17)
    at EventEmitter.<anonymous> (/usr/share/atom/resources/electron.asar/browser/rpc-server.js:367:7)
    at EventEmitter.emit (events.js:182:13)
    at WebContents.<anonymous> (/usr/share/atom/resources/electron.asar/browser/api/web-contents.js:294:13)
    at WebContents.emit (events.js:182:13)


```
At /usr/share/atom/resources/electron.asar/renderer/api/remote.js:219

Error: Cannot call function 'getPosition' on missing remote object 16
Error: Cannot call function 'getPosition' on missing remote object 16
    at throwRPCError (/usr/share/atom/resources/electron.asar/browser/rpc-server.js:145:17)
    at EventEmitter.<anonymous> (/usr/share/atom/resources/electron.asar/browser/rpc-server.js:367:7)
    at EventEmitter.emit (events.js:182:13)
    at WebContents.<anonymous> (/usr/share/atom/resources/electron.asar/browser/api/web-contents.js:294:13)
    at WebContents.emit (events.js:182:13)

    at metaToException (/usr/share/atom/resources/electron.asar/renderer/api/remote.js:271:17)
    at Object.exception (/usr/share/atom/resources/electron.asar/renderer/api/remote.js:219:30)
    at metaToValue (/usr/share/atom/resources/electron.asar/renderer/api/remote.js:223:28)
    at Object.remoteMemberFunction (/usr/share/atom/resources/electron.asar/renderer/api/remote.js:123:16)
    at AtomEnvironment.getWindowDimensions (/usr/share/atom/resources/app/static/<embedded>:1:705395)
    at AtomEnvironment.storeWindowDimensions (/usr/share/atom/resources/app/static/<embedded>:1:705810)
    at WindowEventHandler.handleWindowBlur (/usr/share/atom/resources/app/static/<embedded>:11:285354)
```

### Commands

```
  3x -1:34.3.0 intentions:highlight (input.hidden-input)
     -1:29.8.0 core:close (input.hidden-input)
     -1:19.8.0 intentions:highlight (input.hidden-input)
```

### Non-Core Packages

```
atom-clock 0.1.17 
better-language-sml 1.7.0 
busy-signal 2.0.1 
contrast-syntax 0.8.1 
dark-bint-syntax 0.10.0 
fega-high-contrast-syntax 2.0.1 
file-icons 2.1.34 
file-types 1.0.1 
GAS-x86_64-highlighter 0.3.7 
gl-dark-syntax 3.0.0 
gl-light-syntax 3.0.0 
hard-wrap 1.7.1 
highlight-selected 0.16.0 
intentions 1.1.5 
language-autotools 2.0.1 
language-bison 1.0.0 
language-cmake 1.2.0 
language-common-lisp 0.9.5 
language-diff 0.7.0 
language-docker 1.1.8 
language-gnuplot-5 0.4.2 
language-haskell 1.19.1 
language-latex 1.2.0 
language-lex-flex-new 0.2.1 
language-llvm-ir 1.1.1 
language-mediawiki 2.2.0 
language-ocaml 1.9.5 
language-protobuf 0.7.1 
language-rust 0.4.12 
language-scala 1.1.10 
language-scheme 1.4.0 
latex 0.50.2 
linter 2.3.1 
linter-js-yaml 1.4.6 
linter-json-lint 0.1.3 
linter-spell 0.15.0 
linter-spell-latex 0.11.0 
linter-ui-default 1.7.1 
merge-conflicts 1.4.5 
minimap 4.29.9 
one-dark-vivid-syntax 1.8.1 
pdf-view 0.71.0 
sort-lines 0.19.0 
svn 0.0.14 
tabs-to-spaces 1.0.5 
tree-view-scope-lines 0.2.0 
```

**Comment by andrew-davie** [comment]
I've had this error several times in the last week. For me, it popped up when I did a build and ran an external program. The error came up pretty much at the point the external program was run, so it was in the background, so to speak, as I had the external program in foreground. Once it started showing up, it always showed up. Every build. I had to exit Atom and re-run to clear the problem.

**Comment by lee-dohm** [comment]
@andrew-davie You should probably report the error to the package you use to run the external program then. If you can't reproduce the problem in [Safe Mode](https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode), then the error is virtually always caused by a package you have installed.

**Comment by andrew-davie** [comment]
> @andrew-davie You should probably report the error to the package you use to run the external program then. If you can't reproduce the problem in [Safe Mode](https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode), then the error is virtually always caused by a package you have installed.

I did report it, and it was closed as a duplicate of this bug. So I posted here so that at least the fact that there was another with this issue was known to the OP.

**Comment by lee-dohm** [comment]
Can you reproduce the problem in Safe Mode?

**Comment by andrew-davie** [comment]
> Can you reproduce the problem in Safe Mode?

If that was directed at me... since the problem happens very rarely, perhaps once a week (and only in the last week, at that), there's not much point trying to reproduce it. All you'd get would be a false negative.

**Comment by codingbunnys** [comment]
[Enter steps to reproduce:]

1. Start Atom
2. See Error

**Atom**: 1.40.1 x64
**Electron**: 3.1.10
**OS**: Microsoft Windows 10 Pro
**Thrown From**: Atom Core


### Stack Trace

Uncaught Error: Cannot call function 'getPosition' on missing remote object 27
Error: Cannot call function 'getPosition' on missing remote object 27
    at throwRPCError (C:\Users\PrivateName\AppData\Local\atom\app-1.40.1\resources\electron.asar\browser\rpc-server.js:145:17)
    at EventEmitter.<anonymous> (C:\Users\PrivateName\AppData\Local\atom\app-1.40.1\resources\electron.asar\browser\rpc-server.js:367:7)
    at EventEmitter.emit (events.js:182:13)
    at WebContents.<anonymous> (C:\Users\PrivateName\AppData\Local\atom\app-1.40.1\resources\electron.asar\browser\api\web-contents.js:294:13)
    at WebContents.emit (events.js:182:13)


```
At C:\Users\PrivateName\AppData\Local\atom\app-1.40.1\resources\electron.asar\renderer\api\remote.js:219

Error: Cannot call function 'getPosition' on missing remote object 27
Error: Cannot call function 'getPosition' on missing remote object 27
    at throwRPCError (~/AppData/Local/atom/app-1.40.1/resources/electron.asar/browser/rpc-server.js:145:17)
    at EventEmitter.<anonymous> (~/AppData/Local/atom/app-1.40.1/resources/electron.asar/browser/rpc-server.js:367:7)
    at EventEmitter.emit (events.js:182:13)
    at WebContents.<anonymous> (~/AppData/Local/atom/app-1.40.1/resources/electron.asar/browser/api/web-contents.js:294:13)
    at WebContents.emit (events.js:182:13)

    at metaToException (~/AppData/Local/atom/app-1.40.1/resources/electron.asar/renderer/api/remote.js:271:17)
    at Object.exception (~/AppData/Local/atom/app-1.40.1/resources/electron.asar/renderer/api/remote.js:219:30)
    at metaToValue (~/AppData/Local/atom/app-1.40.1/resources/electron.asar/renderer/api/remote.js:223:28)
    at Object.remoteMemberFunction (~/AppData/Local/atom/app-1.40.1/resources/electron.asar/renderer/api/remote.js:123:16)
    at AtomEnvironment.getWindowDimensions (~/AppData/Local/atom/app-1.40.1/resources/app/static/<embedded>:1:731570)
    at AtomEnvironment.storeWindowDimensions (~/AppData/Local/atom/app-1.40.1/resources/app/static/<embedded>:1:731985)
    at WindowEventHandler.handleWindowBlur (~/AppData/Local/atom/app-1.40.1/resources/app/static/<embedded>:11:285354)
```

### Commands

```
```

### Non-Core Packages

```
language-batch 0.4.0 
language-batchfile 0.7.5 
language-vbscript 0.9.0 
```

**Comment by sadick254** [comment]
Thank you for opening this issue. This was issue present in electron versions 4 and below. The latest atom version `v1.51.0` uses electron version `5.0.3`. The issue was fixed on this commit https://github.com/electron/electron/commit/a8f2646ba682d65f94ea628ac11b1a5521bcedcd. The fix has been available from electron version `5.0.0`. I would advice you to upgrade to the latest atom version. I am closing this for now but happy to reopen if you still experience the issue.
